contracts/vortex/CarbonVortex.sol

```diff
@@ -105,31 +172,171 @@ contract CarbonVortex is ICarbonVortex, Upgradeable, ReentrancyGuardUpgradeable,
+    }
+
+    /**
+     * @notice sets the price decay half-life for all tokens except target
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function setPriceDecayHalfLife(
+        uint32 newPriceDecayHalfLife
+    ) external onlyAdmin greaterThanZero(newPriceDecayHalfLife) {
+        _setPriceDecayHalfLife(newPriceDecayHalfLife);
+    }
+
+    /**
+     * @notice sets the price decay half-life for the target token
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function setTargetTokenPriceDecayHalfLife(
+        uint32 newPriceDecayHalfLife
+    ) external onlyAdmin greaterThanZero(newPriceDecayHalfLife) {
+        _setTargetTokenPriceDecayHalfLife(newPriceDecayHalfLife);
+    }
+
+    /**
+     * @notice sets the price decay half-life for the target token on reset
+     *
+     * Requirements:
+     *
+     * - The caller must be the admin of the contract.
+     */
+    function setTargetTokenPriceDecayHalfLifeOnReset(
+        uint32 newPriceDecayHalfLife
+    ) external onlyAdmin greaterThanZero(newPriceDecayHalfLife) {
+        _setTargetTokenPriceDecayHalfLifeOnReset(newPriceDecayHalfLife);
+    }
+
+    /**
+     * @notice sets the max (or initial) target token sale amount
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function setMaxTargetTokenSaleAmount(
+        uint128 newMaxTargetTokenSaleAmount
+    ) external onlyAdmin greaterThanZero(newMaxTargetTokenSaleAmount) {
+        _setMaxTargetTokenSaleAmount(newMaxTargetTokenSaleAmount);
+    }
+
+    /**
+     * @notice sets the min target token sale amount
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function setMinTargetTokenSaleAmount(
+        uint128 newMinTargetTokenSaleAmount
+    ) external onlyAdmin greaterThanZero(newMinTargetTokenSaleAmount) {
+        _setMinTokenSaleAmount(_targetToken, newMinTargetTokenSaleAmount);
+    }
+
+    /**
+     * @notice sets if trading is enabled or disabled for a token
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function disablePair(Token token) external onlyAdmin {
+        _setPairDisabled(token);
+    }
+
+    /**
+     * @dev withdraws funds held by the contract and sends them to an account
+     *
+     * requirements:
+     *
+     * - the caller must be the admin of the contract
+     */
+    function withdrawFunds(
+        Token[] calldata tokens,
+        address payable target,
+        uint256[] calldata amounts
+    ) external validAddress(target) validateTokens(tokens) nonReentrant onlyAdmin {
+        uint256 len = tokens.length;
+        if (len != amounts.length) {
+            revert InvalidAmountLength();
+        }
+        for (uint256 i = 0; i < len; i = uncheckedInc(i)) {
```

Comment:
It's a good opportunity to upgrade to Solidity 0.8.23 (since it's going to be a new deployment) and also avoid `uncheckedInc` in this contract